docs/migration-guide-NIO1-to-NIO2.md

```diff
@@ -0,0 +1,71 @@
+
+For macOS, download Xcode 10, beta 3 or better and you're good to go. On Linux, use a very recent Swift 5.0 snapshot.
+
+## Step 2: Update your NIO dependencies
+
+Update the NIO packages you use to include the following.
+
+---
+**NOTE**
+
+At this moment, NIO 2 is still in development so if you want to depend on the NIO 2 family packages before we tag any versions, it's `.branch("master")` for all packages.
+
+---
+
+- `swift-nio`: `.package(url: "https://github.com/apple/swift-nio.git", from: "2.0.0")`
+- `swift-nio-ssl`: `.package(url: "https://github.com/apple/swift-nio-ssl.git", from: "2.0.0")`
+- `swift-nio-extras`: `.package(url: "https://github.com/apple/swift-nio.git", from: "1.0.0")`
+- `swift-nio-http2`: `.package(url: "https://github.com/apple/swift-nio-http2.git", from: "1.0.0")`
+- `swift-nio-transport-services`: `.package(url: "https://github.com/apple/swift-nio-transport-services.git", from: "1.0.0")`
+
+After you change your `Package.swift`, run `swift package update`.
+
+## Step 3: `ctx` to `context`
+
+The NIO community has [correctly](https://swift.org/documentation/api-design-guidelines/#avoid-abbreviations) [pointed](https://github.com/apple/swift-nio/issues/663#issuecomment-442013880) [out](https://github.com/apple/swift-nio/issues/483) that the use of `ctx` for the `ChannelHandlerContext` variables feels alien in Swift. We agree with this and have decided that despite the breakage it will cause, we shall rename all mentions of `ctx` to `context`. Unfortunately, that means that every `ChannelHandler` in existence needs a change. To make it worse, `ChannelHandler`s have default implementations for all events which means if you have a `ChannelHandler` method that still uses `ctx`, your code will still compile but the method will never be called because its first parameter is now named `context`.
+
+Therefore, do yourself a favour and start the conversion from NIO 1 to NIO 2 by running the following command in the root of your package:
+
+```bash
+# if you're developing on macOS / BSD
+find . -name '*.swift' -type f | while read line; do sed -E -i '' 's/([^"])[[:<:]]ctx[[:>:]]([^"])/\1context\2/g' "$line"; done
+```
+
+```bash
+# if you're developing on Linux
+find . -name '*.swift' -type f | while read line; do sed -E -i 's/([^"])\<ctx\>([^"])/\1context\2/g' "$line";  done
+```
+
+What these unwieldy commands do is replace `ctx` if it appears as its own word with `context` in all `.swift` files under the current directory. For the NIO packages themselves that was all that was needed for this huge change.
+
+## Step 4: Use `_NIO1APIShims`
+
+The goal of the `_NIO1APIShims` module (which is _not_ public API and is also untested) is to help you with all the small renames we have done for NIO 2. We recommend to follow these steps to make use of as many automatisms as possible:
```

Comment:
Nit: the odds of automatism being a word in most English speaker's vocabularies is approximately zero. Consider rewording to "to make as much use of automation as possible"